_blogposts/compiler/overview-of-new_encoding.mdx

```diff
@@ -0,0 +1,229 @@
+---
+author: hongbo
+date: "2020-06-10 18:00" 
+previewImg:
+category: compiler
+title: Overview of new encoding in BuckleScript
+description: |
+  Highlights of our newest changes to the internal representation of lazy values
+  and how it will benefit our users.
+---
+
+
+---
+title: An overview of new data representation in BuckleScript
+---
+
+In the next version of BuckleScript, we made several major changes to tweak the data representation for various data types, it's more idiomatic and debugger friendly. 
+
+Note since V8 or other JavaScript engines are tweaked to make idiomatic JS code running fast, so it also results in faster running code. 
+
+Another property, for a compiled language like BuckleScript, is that we can reason about [IC](https://en.wikipedia.org/wiki/Inline_caching) friendliness by just looking at the type definitions locally; this is very helpful for advanced users to write performance reliable JS code, if you are unfamiliar with how IC works in general, feel free to skip the section about IC.
```

Comment:
...performant and reliable JS code. If you are unfamiliar...

Reply:
Here I meant `performance predictable`